datafusion/core/src/datasource/file_format/mod.rs

```diff
@@ -58,6 +74,15 @@ pub trait FileFormat: Send + Sync + fmt::Debug {
     /// downcast to a specific implementation.
     fn as_any(&self) -> &dyn Any;
 
+    /// Returns the extension for this FileFormat, e.g. "file.csv" -> csv
+    fn get_ext(&self) -> String;
```

Comment:
Minor: Is there any reason this has to return an owned `String`? Could it return `&str` ?

Reply:
I think &str would be too restrictive, but we could avoid requiring a String allocation with a cow instead.